src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReflectionFieldMapNode.cs

```diff
@@ -43,6 +43,37 @@ public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
 
         protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
 
+        /// <summary>
+        /// Helper method to compute the dependencies that would be needed for reflection field lookup.
+        /// </summary>
+        public static DependencyList GetReflectionFieldMapEntryDependencies(NodeFactory factory, TypeDesc type)
+        {
+            DependencyList dependencies = new DependencyList();
```

Comment:
Nit: We could get rid of these extra allocations by following the existing patterns in e.g. `AddDependenciesDueToInvokeTemplatePresence` or `AddDependenciesDueToReflectability`.